### DISCUSSION TITLE
Configure Inverter Max Output Power

### DISCUSSION THREAD

**Discussion by mmiguel4** [wlcrs/huawei_solar#72] [General]
Is it possible to make Inverter max output power configurable through this integration? I would like to make some kind to near zero grid injection if possible...

**Comment by rowbox** [comment]
I have same need,  to prevent export to grid. In SUN 2000 app Settings > Power Adjustments > Maximum Active Power.
Thanks

**Comment by gugaspinto** [comment]
Hello, its important to me too.
I requested in #20 but it was closed. It will be great if it can be implemented. Thanks everyone.

**Reply by mmiguel4** [reply]
I tried it manually with fusion solar and it works fine. Used the option Active Power derating in absolute value(W).
It was nice have access to it in home assistant.

**Comment by mmiguel4** [comment]
Parameter Setting by Fixed Value
The host delivers the active scheduling target value, which is an absolute value measured in kW. The adjustment precision is 0.1 kW. The maximum value is the reported Pmax, and the minimum value is 0.
Operation:
Change the value of Active power control (register 40118) to 2, and then change the value of Fixed active power derating (register 40120) to the target value (kW).

For more details, please see the brochure

**Reply by wlcrs** [reply]
Where does this text come from? The register 40118 is not mentioned in the 'Solar Inverter Modbus Interface Definitions', dated 2021-08-12. Do you have a more recent version? 

As stated in #20: I don't feel confident in implementing features which are not officially documented and which I don't fully understand myself. I don't have a test setup that I can play with, and don't want to f*ck up my installation by writing undocumented values to undocumented registers.

**Reply by mmiguel4** [reply]
I don't have the document, don't have permission to download from huawei.
Found the info here:
https://forum.huawei.com/enterprise/en/set-active-power-limitation-via-modbus/thread/766121-100027

Also found this, 
https://www.photovoltaikforum.com/core/attachment/87214-sun2000-v200r002-modbus-interface-definitions-pdf/ 

Hope it helps.
I don't mind to try, but I'm kinda new to Home Assisant, modbus,... don't really know where to start. Would be great to have some guidance.

**Comment by mmiguel4** [comment]
Where is the code that sends and receives modbus commands in this integration? Cant't find it... :\

**Reply by wlcrs** [reply]
You can find the library doing the actual Modbus calls here: https://gitlab.com/Emilv2/huawei-solar/-/tree/develop
